app/forms/mappings/Mappings.scala

```diff
@@ -26,6 +26,9 @@ trait Mappings extends Formatters with Constraints {
   protected def text(errorKey: String = "error.required", args: Seq[String] = Seq.empty): FieldMapping[String] =
     of(stringFormatter(errorKey, args)(identity))
 
+  protected def textWithSpacesRemoved(errorKey: String = "error.required", args: Seq[String] = Seq.empty): FieldMapping[String] =
+    of(spacelessStringFormatter(errorKey, args))
```

Comment:
Elsewhere we have defined something like this as `adaptedText` with a function we pass in. So in the form itself you'd have `"value" -> adaptedText(s"$prefix.error.required", args = args.map(_.toString))(_.removeSpaces())`

Reply:
Basically means we don't need to define a bunch of different mappings and formatters that all fundamentally do the same thing